lib/sidekiq/xaclient.rb

```diff
@@ -0,0 +1,52 @@
+require "sidekiq/client"
+
+begin
+  require "after_commit_everywhere"
+rescue LoadError
+  Sidekiq.logger.error("You need to add after_commit_everywhere to your Gemfile for this to work")
+  exit(-127)
+end
+
+module Sidekiq
+  module XAClient
+    include AfterCommitEverywhere
+
+    ##
+    # Control job push within ActiveRecord transactions. Jobs can specify an
+    # "xa" attribute to define the policy to use:
+    #   * true or "commit" means enqueue the job after committing the current transaction.
+    #   * "rollback" means enqueue this job only if the current transaction rolls back
```

Comment:
Does anybody really need this behaviour? I read https://github.com/mperham/sidekiq/issues/5239#issuecomment-1064356789. I'm not really convinced by it. Moreover, I understand

> Under those circumstances, it can be useful to push a job to Sidekiq that will run even if the transaction fails.

as "enqueue this job after the current transaction finishes - either commits or rolls back".

What did you mean @ianhall?